File: typescript-client-sdk/src/sdk/models/shared/codatdatacontractsdatasetscommercedisputepagedresponsemodel.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { CodatDataContractsDatasetsCommerceDispute } from "./codatdatacontractsdatasetscommercedispute";
import { CodatDataContractsDatasetsCommerceDisputePagedResponseLinksModel } from "./codatdatacontractsdatasetscommercedisputepagedresponselinksmodel";
import { Expose, Type } from "class-transformer";


// CodatDataContractsDatasetsCommerceDisputePagedResponseModel
/** 
 * Used to represent what can be returned by an endpoint that supports paging.
 * Usable with the [ProducesResponseType] attribute on a controller action.
**/
export class CodatDataContractsDatasetsCommerceDisputePagedResponseModel extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "_links" })
  @Type(() => CodatDataContractsDatasetsCommerceDisputePagedResponseLinksModel)
  links?: CodatDataContractsDatasetsCommerceDisputePagedResponseLinksModel;

  @SpeakeasyMetadata()
  @Expose({ name: "pageNumber" })
  pageNumber?: number;

  @SpeakeasyMetadata()
  @Expose({ name: "pageSize" })
  pageSize?: number;

  @SpeakeasyMetadata({ elemType: CodatDataContractsDatasetsCommerceDispute })
  @Expose({ name: "results" })
  @Type(() => CodatDataContractsDatasetsCommerceDispute)
  results?: CodatDataContractsDatasetsCommerceDispute[];

  @SpeakeasyMetadata()
  @Expose({ name: "totalResults" })
  totalResults?: number;
}

File: go-client-sdk/pkg/models/shared/codatdatacontractsdatasetscommercedisputepagedresponsemodel.go
package shared

// CodatDataContractsDatasetsCommerceDisputePagedResponseModel
// Used to represent what can be returned by an endpoint that supports paging.
// Usable with the [ProducesResponseType] attribute on a controller action.
type CodatDataContractsDatasetsCommerceDisputePagedResponseModel struct {
	Links        *CodatDataContractsDatasetsCommerceDisputePagedResponseLinksModel `json:"_links,omitempty"`
	PageNumber   *int                                                              `json:"pageNumber,omitempty"`
	PageSize     *int                                                              `json:"pageSize,omitempty"`
	Results      []CodatDataContractsDatasetsCommerceDispute                       `json:"results,omitempty"`
	TotalResults *int                                                              `json:"totalResults,omitempty"`
}
